internal/keys/systemstore_field_id.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package keys
+
+import (
+	"strconv"
+	"strings"
+
+	ds "github.com/ipfs/go-datastore"
+)
+
+// FieldID indexes field short ids by the full id.
+type FieldID struct {
+	CollectionShortID uint32
+	FieldID           string
+}
+
+var _ Key = (*SchemaRootKey)(nil)
+
+func NewFieldID(collectionShortID uint32, fieldID string) FieldID {
+	return FieldID{
+		CollectionShortID: collectionShortID,
+		FieldID:           fieldID,
+	}
+}
+
+func NewFieldIDP(collectionShortID uint32) FieldID {
```

Comment:
suggestion: This function is confusing. Probably mostly because of the name. It makes you wonder what `P` stands for. Either `NewFieldIDPrefix` or just using `NewFieldID(1, "")` would be better.

Reply:
I meant to change this and then forgot about this, and missed it scanning through :) Will change.

- [ ] `NewFieldIDP`